src/main.rs

```diff
@@ -1,14 +1,51 @@
-use std::io::Read;
+use schemars::schema_for;
+use std::{env::args, fs, io::Read, process::exit};
 
 use nftables::schema::Nftables;
 
 fn main() {
-    deserialize_stdin();
+    // Get command arguments.
+    // We can skip the first argument in any case, because its the program path itself
+    let args = args().skip(1).collect();
+
+    handle_args(args);
+}
+
+fn handle_args(args: Vec<String>) {
+    let mut args = args.into_iter();
+    match args.next() {
+        Some(command) if command == "schema" => {
+            generate_json_schema(args.next().unwrap_or("./nftables.schema.json".to_string()));
+        }
+        Some(command) => {
+            println!("Unknown command: `{command}`. Try with schema to generate a Json schema or call with stdin only.");
+            exit(1);
+        }
+        // Called without any arguments so do what we did before already
+        None => {
+            deserialize_stdin();
+        }
+    }
+}
+
+fn generate_json_schema(schema_dst_path: String) {
+    let schema = schema_for!(Nftables);
+
+    if let Err(err) = fs::write(
+        schema_dst_path.clone(),
+        serde_json::to_string_pretty(&schema).unwrap(),
+    ) {
+        println!("Failed to write data to file: {err}");
+        exit(1);
+    }
+
+    println!("Wrote schema data to: {schema_dst_path}");
 }
 
```

Comment:
The documentation from above could be moved into a doc comment for this function.

```suggestion

/// Deserializes nftables JSON from the standard input and prints the result.
///
/// This is the default behavior when the executable is called without any
/// arguments.
```